test/unit/DCAStrategies/dca-strategies-management-handler.spec.ts

```diff
@@ -0,0 +1,106 @@
+  });
+
+  describe('createStrategy', () => {
+    const OWNER = wallet.generateRandomAddress();
+    const NAME = ethers.utils.formatBytes32String('Optimism Ecosystem - v1');
+    const SHARE_TOKEN_A = { token: wallet.generateRandomAddress(), share: BigNumber.from(50) };
+    const SHARE_TOKEN_B = { token: wallet.generateRandomAddress(), share: BigNumber.from(50) };
+    const SHARES = [SHARE_TOKEN_A, SHARE_TOKEN_B];
+
+    when('owner is zero address', () => {
+      then('tx reverted with message', async () => {
+        await expect(DCAStrategiesManagementHandlerMock.createStrategy(NAME, SHARES, constants.ZERO_ADDRESS)).to.be.revertedWith(
+          'ZeroAddress()'
+        );
+      });
+    });
+    when('strategy is created', () => {
+      let tx: TransactionResponse;
+      let strategy: IDCAStrategiesManagementHandler.StrategyStruct;
+      let tokenShares: IDCAStrategies.ShareOfTokenStruct[];
+      given(async () => {
+        tx = await DCAStrategiesManagementHandlerMock.createStrategy(NAME, SHARES, OWNER);
+        strategy = await DCAStrategiesManagementHandlerMock.getStrategy(1);
+        tokenShares = strategy.tokens;
+      });
+      when('name already exists', () => {
+        then('tx reverted with message', async () => {
+          await expect(DCAStrategiesManagementHandlerMock.createStrategy(NAME, SHARES, OWNER)).to.be.revertedWith('NameAlreadyExists()');
+        });
+      });
+      then('counter is correct', async () => {
+        expect(await DCAStrategiesManagementHandlerMock.strategyCounter()).to.be.equal(1);
+      });
+      then('owner is correct', async () => {
+        expect(strategy.owner).to.be.equal(OWNER);
+      });
+      then('name is correct', async () => {
+        expect(strategy.name).to.be.equal(NAME);
+      });
+      then('name is saved correctly in reversed mapping', async () => {
+        expect(await DCAStrategiesManagementHandlerMock.strategyIdByName(NAME)).to.be.equal(1);
+      });
+      then('version is correct', async () => {
+        expect(strategy.currentVersion).to.be.equal(1);
+      });
+      then('shares are correct', async () => {
+        SHARES.forEach((s, i) => {
+          expect(tokenShares[i].share).to.be.equal(s.share);
+          expect(tokenShares[i].token).to.be.equal(s.token);
+        });
+      });
+      then('event is emitted', async () => {
+        let _strategyId: BigNumber = await readArgFromEventOrFail(tx, 'StrategyCreated', 'strategyId');
+        let _strategyName: string = await readArgFromEventOrFail(tx, 'StrategyCreated', 'strategyName');
+        let _tokens: IDCAStrategies.ShareOfTokenStruct[] = await readArgFromEventOrFail(tx, 'StrategyCreated', 'tokens');
+        let _owner: string = await readArgFromEventOrFail(tx, 'StrategyCreated', 'owner');
+        expect(_strategyId).to.be.equal(1);
+        expect(_strategyName).to.be.equal(strategy.name);
+        expect(_owner).to.be.equal(strategy.owner);
+        _tokens.forEach((s: any, i: any) => {
```

Comment:
Instead of using `_tokens`, why not compare against `SHARES` like the test before? Also, let's compare the length